testbench/common.py

```diff
@@ -737,10 +738,55 @@ def limited_response_fn(data):
     return limited_response_fn
 
 
+def grpc_handle_retry_test_instruction(database, request, context, method):
+    test_id = None
+    if context is not None:
+        if hasattr(context, "invocation_metadata") and isinstance(
+            context.invocation_metadata(), Iterable  # Handle mocks in tests
+        ):
+            metadata = context.invocation_metadata()
+            if not isinstance(metadata, dict):
+                metadata = dict(metadata)
```

Comment:
What type are you expecting metadata to be here if not a dict already? Most iterables can't be cast to dict.

Reply:
Updated; The invocation metadata is expected to be a tuple.